packages/flutter/lib/src/material/bottom_app_bar_theme.dart

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Defines a theme for [BottomAppBar] widgets.
+///
+/// Descendant widgets obtain the current [BottomAppBarTheme] object using
+/// `BottomAppBarTheme.of(context)`. Instances of [BottomAppBarTheme] can be
+/// customized with [BottomAppBarTheme.copyWith].
+///
+/// See also:
+///
+///  * [BottomAppBar], a material bottom app bar that can be customized using
+///    this [BottomAppBarTheme].
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class BottomAppBarTheme extends Diagnosticable {
+  /// Creates a BAB theme that can be used for [ThemeData.BottomAppBarTheme].
+  const BottomAppBarTheme({
+    this.color,
+    this.elevation,
+    this.shape,
+  });
+
+  /// Default value for [BottomAppBar.color].
+  ///
+  /// If null, [ThemeData.bottomAppBarColor] is used, if that's null, defaults
+  /// to [Colors.white].
+  final Color color;
+
+  /// Default value for [BottomAppBar.elevation].
+  ///
+  /// If null, defaults to `8.0`.
+  final double elevation;
+
+  /// Default value for [BottomAppBar.shape].
+  final NotchedShape shape;
+
+  /// Creates a copy of this object but with the given fields replaced with the
+  /// new values.
+  BottomAppBarTheme copyWith({
+    Color color,
+    double elevation,
+    NotchedShape shape,
+  }) {
+    return BottomAppBarTheme(
+      color: color ?? this.color,
+      elevation: elevation ?? this.elevation,
+      shape: shape ?? this.shape,
+    );
+  }
+
+  /// The data from the closest [BottomAppBarTheme] instance given the build context.
```

Comment:
So I should add something like ```/// Looks up [Theme] of the current context and pulls out the [ThemeData.bottomAppBar]```? 

To me this seems like an implementation detail that is too specific for documentation, especially since we plan to eventually change this implementation of this to *not* look up the `Theme` (or am I wrong about this?)

Reply:
Typically an expression like "the closest" means the nearest ancestor of the context and typically the expression is referring to a widget. 

How about: The [ThemeData.bottomAppBarTheme] property of the ambient [Theme].

I wish it wasn't "ambient", but that's the term of art that's in common usage in our docs.

Reply:
Gotcha, well I will make it consistent!